build-system/tasks/pr-check.js

```diff
@@ -51,27 +48,44 @@ async function prCheck(cb) {
   };
 
   const startTime = startTimer(FILENAME, FILENAME);
-  const buildTargets = determineBuildTargets();
-  printChangeSummary(FILENAME);
+  if (!runYarnChecks(FILENAME)) {
+    stopTimer(FILENAME, FILENAME, startTime);
+    process.exitCode = 1;
+    return;
+  }
 
-  if (
-    !runYarnChecks(FILENAME) ||
-    !areValidBuildTargets(buildTargets, FILENAME)
-  ) {
-    failTask();
+  printChangeSummary(FILENAME);
+  const buildTargets = new Set();
+  if (!determineBuildTargets(buildTargets, FILENAME)) {
+    stopTimer(FILENAME, FILENAME, startTime);
+    process.exitCode = 1;
+    return;
   }
 
   runCheck('gulp lint --local-changes');
   runCheck('gulp presubmit');
-  runCheck('gulp ava');
-  runCheck('gulp babel-plugin-tests');
-  runCheck('gulp caches-json');
-  runCheck('gulp json-syntax');
+
```

Comment:
Possible refactor - use `pr-check/checks.js` and others to provide build target logic here.

Reply:
Good idea. I didn't do it here because I wanted to avoid PR-bloat like last week :smiley: